utils/nvme.go

```diff
@@ -141,161 +139,192 @@ func (n *Nvme) cmdListCapabilities(ctx context.Context, logicalPath string) ([]b
-				continue
-			}
-
-			data := strings.Split(parts[2], "\t")
-			enabled := strings.TrimSpace(data[0])
-
-			if enabled != "0" {
-				capability.Enabled = true
-			}
-
-			// Generate short flag identifier
-			for _, word := range strings.Fields(data[1]) {
-				capability.Name += strings.ToLower(word[0:1])
-			}
-
-			capability.Description = data[1]
+	var capabilitiesFound []*common.Capability
+	capabilitiesFound = append(capabilitiesFound, parseSanicap(caps.SANICAP)...)
+	capabilitiesFound = append(capabilitiesFound, parseFna(caps.FNA)...)
 
-			if enabled != "0" {
-				capability.Enabled = true
-			}
+	return capabilitiesFound, nil
+}
 
-			capability.Description = data[1]
-			capabilitiesFound = append(capabilitiesFound, capability)
-			// sanitize
-		case (saniBool && !saniEnd && !isBlank):
-			capability := new(common.Capability)
+// nolint:gomnd // bit twiddling and magic numbers go hand in hand for legibility
+func parseFna(fna uint) []*common.Capability {
+	caps := make([]*common.Capability, 0, 4)
+
+	caps = append(caps, &common.Capability{
+		Name:        "fna",
+		Description: "Crypto Erase Support",
+		Enabled:     fna != 0,
+	})
+
+	// nvme: cese
+	if (fna&0b100)>>2 == 0 {
+		caps = append(caps, &common.Capability{
+			Name:        "censapose",
+			Description: "Crypto Erase Not Supported as part of Secure Erase",
+			Enabled:     false,
+		})
+	} else {
+		caps = append(caps, &common.Capability{
+			Name:        "cesapose",
+			Description: "Crypto Erase Supported as part of Secure Erase",
+			Enabled:     true,
+		})
+	}
 
-			var partsLen = 3
+	// nvme: cens
+	if (fna&0b10)>>1 == 0 {
+		caps = append(caps, &common.Capability{
+			Name:        "ceatsn",
+			Description: "Crypto Erase Applies to Single Namespace(s)",
+			Enabled:     false,
+		})
+	} else {
+		caps = append(caps, &common.Capability{
+			Name:        "ceatan",
+			Description: "Crypto Erase Applies to All Namespace(s)",
+			Enabled:     true,
+		})
+	}
 
-			parts := strings.Split(line, ":")
-			if len(parts) != partsLen {
-				continue
-			}
+	// nvme: fmns
+	if (fna&0b1)>>0 == 0 {
+		caps = append(caps, &common.Capability{
+			Name:        "fatsn",
+			Description: "Format Applies to Single Namespace(s)",
+			Enabled:     false,
+		})
+	} else {
+		caps = append(caps, &common.Capability{
+			Name:        "fatan",
+			Description: "Format Applies to All Namespace(s)",
+			Enabled:     true,
+		})
+	}
 
-			data := strings.Split(parts[2], "\t")
-			enabled := strings.TrimSpace(data[0])
+	return caps
+}
 
-			if enabled != "0" {
-				capability.Enabled = true
-			}
+// nolint:gomnd // bit twiddling and magic numbers go hand in hand for legibility
+func parseSanicap(sanicap uint) []*common.Capability {
+	caps := make([]*common.Capability, 0, 6)
+
+	caps = append(caps, &common.Capability{
+		Name:        "sanicap",
+		Description: "Sanitize Support",
+		Enabled:     sanicap != 0,
+	})
+
+	// nvme: nodmmas
+	switch (sanicap & (0b11 << 30)) >> 30 {
```

Comment:
https://github.com/linux-nvme/nvme-cli/blob/v2.8/nvme-print-stdout.c#L2064-L2093.

Upstream uses hex instead of bit 0b and bit shifts. I opted for binary notation and bit shifts to make it obvious which bits we're interested. Here we're interested in 2 bits, bits 32 and 31.

Reply:
fixed, PTAL